src/cryptography/x509.py

```diff
@@ -104,6 +104,27 @@ def __repr__(self):
     dotted_string = utils.read_only_property("_dotted_string")
 
 
+class Name(object):
+    def __init__(self, attributes):
+        self._attributes = attributes
+
+    def get_attributes_for_oid(self, oid):
+        return [i for i in self._attributes if i.oid == oid]
+
+    @property
+    def attributes(self):
+        return self._attributes[:]
```

Comment:
There aren't alternate representations of this data (other than filtering it via `get_attributes_for_oid` or similar) so I suppose a property isn't required. No objection.